library/std/src/os/uefi/env.rs

```diff
@@ -0,0 +1,45 @@
+//! UEFI-specific extensions to the primitives in `std::env` module
+
+use crate::ffi::c_void;
+use crate::ptr::NonNull;
+use crate::sync::atomic::{AtomicPtr, Ordering};
+use crate::sync::Once;
+
+static GLOBAL_SYSTEM_TABLE: AtomicPtr<c_void> = AtomicPtr::new(crate::ptr::null_mut());
+static GLOBAL_IMAGE_HANDLE: AtomicPtr<c_void> = AtomicPtr::new(crate::ptr::null_mut());
+pub(crate) static GLOBALS: Once = Once::new();
```

Comment:
Well, I'm not sure if using `NonNull` instead of `AtomicPtr` is correct here. It is technically possible for some other driver to change the underlying System Table and thus atomic operations seem to make more sense to me.

Reply:
I'm not sure I catch your meaning here? `NonNull` does not mandate that the underlying data stays constant, so it should be fine to use even if that situation arises.

Reply:
I was referring to the `AtomicPtr::load()` operation since there will be multiple pointers to SystemTable floating around even in Rust code simultaneously.